tasks/backfill_test_instances.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+from shared.django_apps.reports.models import TestInstance
+from sqlalchemy import desc
+
+from app import celery_app
+from celery_config import backfill_test_instances_task_name
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class BackfillTestInstancesTask(
+    BaseCodecovTask, name=backfill_test_instances_task_name
+):
+    def run_impl(self, *args, dry_run=True, **kwargs):
+        log.info(
+            "Updating test instances",
+        )
+
+        test_instance_filter = TestInstance.objects.select_related(
+            "upload__report__commit"
+        ).filter(
+            branch=None,
+            commitid=None,
+        )
+        num_test_instances = test_instance_filter.count()
+        all_test_instances = test_instance_filter.all()
```

Comment:
Depending on how many test instances we have this could lead to OOM errors, right? I think there are millions of test instances, after all.
Consider using the `yield_per` here, so we don't get everything into memory at once.

Reply:
You can use the `yield_per` value as the chunk size and simplify the code a bit.

Reply:
Interesting. Today I learned.